### ISSUE TITLE
Support for multiple subscriptions to a single CW Log Group

### ISSUE THREAD

**Issue by rjswitzer3** [serverless/serverless#8276]
### Use case description

The ability to have multiple subscriptions (i.e.  by two or more discrete Serverless Applications) to a single CW Log Group.

Right now, if a CW Log Group is declared as the event source in two discrete Serverless applications, whichever application's deployment precedes the other will have its CW Log Group event/trigger removed from the lambdas.

For example: Given two discrete Serverless applications - application A and application B - the CW Log Group event/trigger shall  be applied to the lambda's of application A accordingly upon deployment. Once the subsequent deployment of application B has completed, the event/trigger will have been applied to application B's lambdas accordingly and removed from application A's. Executing the deployment of application A after the fact will not will not re-associate the event/trigger with application A's lambdas, and will not remove them from application B's lambdas. Only after removing application A and then re-deploying it will the effect be observed again.

The CW Log Group trigger can be re-applied to the lambdas through the console if removed, but this is not a sustainable solution. There are other work arounds for this, but it would make more sense to have this as an available feature rather than creating more infrastructure to act as a shim.

**Comment by medikoo** [comment]
@rjswitzer3 Limit of just one subscriber is on AWS side. I believe you should post this request to AWS support team. On Framework level we cannot do anything about that.

**Comment by rjswitzer3** [comment]
@medikoo I shall open a feature request ticket with AWS and follow up with the response I get.

What I find to be peculiar is that you have the ability to subscribe multiple lambdas to a single log group via the console, despite the CloudWatch user guide stating:
> Each log group can have only one subscription filter associated with it. 

I've heard rumor that support for multiple subscriptions were coming this month (i.e. September 2020), but I have yet to find a formal announcement on AWS's behalf.

More to come...

**Comment by medikoo** [comment]
Thanks @rjswitzer3 for investigating that. Let us know if you have some more information

**Comment by rjswitzer3** [comment]
@medikoo Here is the response I received from AWS:

> Hello
>
>My name is Wendy and I will assist you today.  I understand that you are requesting to have multiple filter subscriptions made available to a single CW Log Group.
>
>Regarding your observation, " this statement from the user guide does not hold true when using the console, for one is able to manually subscribe multiple lambdas to a single CloudWatch Log Group with different filters", I tested this out in my account and it also worked for me.  I consulted our in house Subject Matter Expert for CloudWatch who advised that it appears that the limits have been increased recently, but the documentation has not yet been updated.
>
>I hope this helps, but please don't hesitate to contact me should you have any further questions.
> ...

TL;DR AWS does now support multiple subscriptions to a single CW Log Group.

It would appear the ball is now in your court.

**Comment by medikoo** [comment]
@rjswitzer3 thanks for checking that out. Can you confirm it's also supported on CloudFormation level?

If it is, then I believe it should work out of a box, if not, then we still can't do much about that I think

**Comment by rjswitzer3** [comment]
@medikoo Here is the follow-up response I received regarding CloudFormation support:
>Hello Ryan
>
>Thanks for your response.
>
>I'm not sure if the feature is supported by CloudFormation yet as it appears to be a recent change, but you may want to check the What's New [1] and Blog [2] pages for any announcements that may be made in this regard.

It sounds like this feature is still in development.

**Comment by dashmug** [comment]
This feature was just announced yesterday. I'm not sure if CloudFormation already supports it. https://aws.amazon.com/about-aws/whats-new/2020/10/amazon-cloudwatch-logs-now-supports-two-subscription-filters-per-log-group/

**Comment by poerty** [comment]
Hi, I made small [PR](https://github.com/serverless/serverless/pull/9760) for this issue.
(I believe im going the way to resolve issue #9528 which really effects my project)
If you dont mind, can you check for it?

**Comment by pgrzesik** [comment]
Thanks @poerty, it's much appreciated :bow:  - it will be checked today.

**Comment by pgrzesik** [comment]
I'm going to re-open this issue as #9760 addresses the described problem partially (it can still remove subscriptions defined outside of the stack) but I know @poerty will be working on addressing that as well :raised_hands:
